cpp/oneapi/dal/algo/dbscan/backend/cpu/compute_kernel_brute_force.cpp

```diff
@@ -0,0 +1,134 @@
+
+#include "oneapi/dal/algo/dbscan/backend/cpu/compute_kernel.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "oneapi/dal/exceptions.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+#include <daal/src/algorithms/dbscan/dbscan_kernel.h>
+
+namespace oneapi::dal::dbscan::backend {
+
+using std::int64_t;
+using dal::backend::context_cpu;
+using descriptor_t = detail::descriptor_base<task::clustering>;
+
+namespace daal_dbscan = daal::algorithms::dbscan;
+namespace interop = dal::backend::interop;
+
+template <typename Float, daal::CpuType Cpu>
+using daal_dbscan_default_dense_t =
+    daal_dbscan::internal::DBSCANBatchKernel<Float, daal_dbscan::defaultDense, Cpu>;
+
+template <typename Float, daal::CpuType Cpu>
+class dbscan_compute_wrapper {
+public:
+    template <typename... Args>
+    auto compute(Args... args) {
+        auto pack = std::tuple<Args...>(args...);
+        const daal_dbscan::Parameter* par = std::get<sizeof...(Args) - 1>(pack);
+        if (par->memorySavingMode == false) {
+            return daal_dbscan::internal::DBSCANBatchKernel<Float, daal_dbscan::defaultDense, Cpu>{}
+                .computeNoMemSave(std::forward<Args>(args)...);
+        }
+        return daal_dbscan::internal::DBSCANBatchKernel<Float, daal_dbscan::defaultDense, Cpu>{}
+            .computeMemSave(std::forward<Args>(args)...);
+    }
+};
+
+template <typename Float, typename Task>
+static compute_result<Task> call_daal_kernel(const context_cpu& ctx,
+                                             const descriptor_t& desc,
+                                             const table& data,
+                                             const table& weights) {
+    const int64_t row_count = data.get_row_count();
+    const int64_t column_count = data.get_column_count();
+
+    const double epsilon = desc.get_epsilon();
+    const int64_t min_observations = desc.get_min_observations();
+
+    daal_dbscan::Parameter par(epsilon, dal::detail::integral_cast<std::size_t>(min_observations));
+    par.memorySavingMode = desc.get_mem_save_mode();
+    par.resultsToCompute = daal_dbscan::computeCoreIndices | daal_dbscan::computeCoreObservations;
+
+    const auto daal_data = interop::convert_to_daal_table<Float>(data);
+    const auto daal_weights = interop::convert_to_daal_table<Float>(weights);
+
+    array<int> arr_responses = array<int>::empty(row_count * 1);
+    array<int> arr_cluster_count = array<int>::empty(1);
+
+    const auto daal_responses = interop::convert_to_daal_homogen_table(arr_responses, row_count, 1);
+    const auto daal_core_observation_indices = interop::empty_daal_homogen_table<int>(1);
+    const auto daal_core_observations = interop::empty_daal_homogen_table<Float>(column_count);
+    const auto daal_cluster_count = interop::convert_to_daal_homogen_table(arr_cluster_count, 1, 1);
```

Comment:
Why do we have different strategies for memory allocation, for `daal_responses`  memory is allocated via array, but for `daal_core_observations` via old numeric tables.

Reply:
daal_core_observations might be resized inside the kernel - pointer to the array might become invalid

Reply:
Due to that, it useless to allocate memory before the kernel call.

Reply:
Added a comment to clarify the difference